src/player.rs

```diff
@@ -18,4 +30,55 @@ pub struct PlayerBundle {
     #[sprite_sheet_bundle("atlas/Lil_Gobbo.png", 32.0, 32.0, 4, 1, 0.0, 0.0, 3)]
     sprite_bundle: SpriteSheetBundle,
     pub ground_detection: GroundDetection,
+    pub movement_state: EntityMovementState,
+}
+
+pub fn movement(
+    input: Res<Input<KeyCode>>,
+    // time: Res<Time>,
+    mut q_velocity: Query<(
+        &mut Velocity, 
+        &GroundDetection, 
```

Comment:
Minor point, but this system runs in parallel with the one that handles ground detection. Might have to schedule these systems in order to avoid 1 frame inconsistencies. Probably no one would notice anyway, but still.